File: 1/mario_more/mario.c
#include <cs50.h>
#include <stdio.h>

// function placeholder
int get_height(string prompt);

int main(void)
{
    int height = get_height("Height: ");
    for (int counter = 0; counter < height; counter++)
    {
        for (int spaces = 0; spaces < height - 1 - counter ; spaces++)
            // print right whitespaces
        {
            printf(" ");
        }
        
        for (int hashs = 0; hashs < counter + 1 ; hashs++)
            // print right hashs
        {
            printf("#");
        }
        // print mid whitespace
        printf("  ");
        for (int hashs = 0; hashs < counter + 1 ; hashs++)
            // print left hashs
        {
            printf("#");
        }
        printf("\n");
    }
}

int get_height(string prompt)
// input must be a positive intger less than 8
{
    int input;
    do
    {
        input = get_int("%s", prompt);
    }
    // only accept the input if it's between 1 and 8
    while (input < 1 || input > 8);
    return input;
}

File: 6/mario_less/mario.py
#!/usr/bin/env python3
# Mario less

from cs50 import get_int


def get_height():
    while True:
        height = get_int("Height: ")
        if 0 < height <= 8:
            return height


def main():
    height = get_height()
    for hashs in range(height):
        for spaces in range(height - 1 - hashs):
            # print right whitespaces
            print(' ', end='')
        for line in range(hashs + 1):
            # print right hashs
            print('#', end='')
        print('\n', end='')


if __name__ == '__main__':
    main()
